### PR TITLE
feat: `#[contract(module=...)]` and `#[messages]` not needed for trait impl

### PR DESCRIPTION
**Pull request by kulikthebird**

### CODE REVIEW

File: sylvia-derive/src/querier.rs [hunk 1]
```diff
@@ -34,133 +36,54 @@ where
...
-                }
-            }
-
-            impl <'a, C: #sylvia ::cw_std::CustomQuery, #(#generics,)* > Querier for BoundQuerier<'a, C, #(#generics,)* > #where_clause {
-                #(#types_definition)*
-
-                #(#methods_impl)*
-            }
+            pub use #sylvia ::types::{BoundQuerier, Remote};
+            use std::marker::PhantomData;
```
Comment (jawoznia) [inline sylvia-derive/src/querier.rs:69]: Don't use imports in generated code. Instead use full path where needed.

File: sylvia-derive/src/querier.rs [hunk 2]
```diff
@@ -201,6 +117,8 @@ impl<'a> ContractQuerier<'a> {
             .iter()
             .map(|variant| variant.emit_querier_impl::<GenericParam>(&api_path, &generics));
 
+        let methods_impl_copy = methods_impl.clone();
```
Comment (jawoznia) [inline sylvia-derive/src/querier.rs:120]:
```suggestion
            .map(|variant| variant.emit_querier_impl::<GenericParam>(&api_path, &generics))
            .collect();
```

File: sylvia-derive/src/querier.rs [hunk 3]
```diff
@@ -215,42 +133,24 @@ impl<'a> ContractQuerier<'a> {
...
-            }
-
-            impl <'a, C: #sylvia ::cw_std::CustomQuery, #(#generics,)* > Querier for BoundQuerier<'a, C, #(#generics,)* > #where_clause {
-                #(#types_implementation)*
-
-                #(#methods_impl)*
-            }
+        quote! {
+            pub use #sylvia ::types::{BoundQuerier, Remote};
+            use std::marker::PhantomData;
```
Comment (jawoznia) [inline sylvia-derive/src/querier.rs:141]: same

File: sylvia/src/types.rs [hunk 4]
```diff
@@ -3,6 +3,81 @@ use cosmwasm_std::{Deps, DepsMut, Empty, Env, MessageInfo};
...
+
+impl<'a, C: cosmwasm_std::CustomQuery, Contract> From<&'a BoundQuerier<'a, C, Contract>>
+    for BoundQuerier<'a, C, Contract>
+{
+    fn from(input: &'a BoundQuerier<'a, C, Contract>) -> Self {
+        BoundQuerier::borrowed(input.contract, input.querier)
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, JsonSchema)]
```
Comment (jawoznia) [inline sylvia/src/types.rs:41]:
```suggestion
#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
```

**Review by jawoznia — COMMENTED** [review]
Some comments. Also I need to think a bit more how this changes the Api.

File: sylvia-derive/src/querier.rs [hunk 5]
```diff
@@ -34,133 +36,54 @@ where
...
+            interface_name,
         } = self;
 
         let generics: Vec<_> = associated_types
             .without_special()
             .map(ItemType::as_name)
             .collect();
-        let methods_impl = variants
+
+        let accoc_types: Vec<_> = associated_types
```
[the remark this answers is not in the document]
Comment (kulikthebird) [reply sylvia-derive/src/querier.rs]: Done

File: sylvia-derive/src/querier.rs [hunk 6]
```diff
@@ -34,133 +36,54 @@ where
...
+            .map(ItemType::as_name)
+            .map(|assoc| quote! {Self:: #assoc})
+            .collect();
+        let methods_trait_impl = variants
             .variants()
             .iter()
-            .map(|variant| variant.emit_trait_querier_impl(&generics));
+            .map(|variant| variant.emit_trait_querier_impl(&accoc_types));
+
+        let methods_trait_impl_copy = methods_trait_impl.clone();
```
[the remark this answers is not in the document]
Comment (kulikthebird) [reply sylvia-derive/src/querier.rs]: Done

File: sylvia-derive/src/querier.rs [hunk 7]
```diff
@@ -34,133 +36,54 @@ where
...
 
         let methods_declaration = variants
             .variants()
             .iter()
             .map(|variant| variant.emit_querier_declaration(&generics));
 
-        let types_declaration = associated_types.filtered();
-        let types_definition = associated_types.emit_types_definition();
+        let types_declaration = associated_types.filtered().cloned();
```
[the remark this answers is not in the document]
Comment (kulikthebird) [reply sylvia-derive/src/querier.rs]: Done

File: sylvia-derive/src/querier.rs (hunk 1 above)
Comment (kulikthebird) [reply sylvia-derive/src/querier.rs]: Done

File: sylvia/src/types.rs (hunk 4 above)
Comment (kulikthebird) [reply sylvia/src/types.rs]: Done

File: sylvia-derive/src/querier.rs (hunk 3 above)
Comment (kulikthebird) [reply sylvia-derive/src/querier.rs]: Done

**Pull request merged**